core/src/buffer.rs

```diff
@@ -0,0 +1,209 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021-2022 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Vector with limited len realization.
+
+use core::{convert::TryFrom, fmt::Display, marker::PhantomData};
+
+use alloc::{vec, vec::Vec};
+use codec::{Decode, Encode};
+use scale_info::TypeInfo;
+
+/// Limited len vector.
+/// `T` is data type.
+/// `E` is overflow error type.
+/// `N` is max len which a vector can have.
+#[derive(Clone, Default, Debug, Eq, Hash, Ord, PartialEq, PartialOrd, Decode, Encode, TypeInfo)]
+pub struct LimitVec<T, E, const N: usize>(Vec<T>, PhantomData<E>);
+
+impl<T, E: Default, const N: usize> TryFrom<Vec<T>> for LimitVec<T, E, N> {
+    type Error = E;
+    fn try_from(x: Vec<T>) -> Result<Self, Self::Error> {
+        if x.len() > N {
+            Err(E::default())
+        } else {
+            Ok(Self(x, Default::default()))
+        }
+    }
+}
+
+impl<T: Clone + Default, E: Default, const N: usize> LimitVec<T, E, N> {
```

Comment:
Since all methods return `Result` it is better to prepend all names with `try_`

Reply:
Done